vsintegration/src/FSharp.ProjectSystem.FSharp/Project.fs

```diff
@@ -1739,49 +1726,66 @@ namespace rec Microsoft.VisualStudio.FSharp.ProjectSystem
                     VSConstants.S_OK
                 member x.UpdateSolution_Cancel() =
                     VSConstants.S_OK
-                member x.OnActiveProjectCfgChange(pIVsHierarchy) =
-                    OnActiveProjectCfgChange(pIVsHierarchy)
+                member x.OnActiveProjectCfgChange(pHierProj) =
+                    UpdateConfig(pHierProj)
+                    VSConstants.S_OK
                 member x.UpdateProjectCfg_Begin(pHierProj, _pCfgProj, _pCfgSln, _dwAction, pfCancel) =
                     UpdateConfig(pHierProj)
                     VSConstants.S_OK
                 member x.UpdateProjectCfg_Done(pHierProj, _pCfgProj, _pCfgSln, _dwAction, _fSuccess, _fCancel) =
                     UpdateConfig(pHierProj)
+                    // We wait until allproject configurations have been updated before updating our MSBuild state
+                    projNode.UpdateMSBuildState()
+                    projNode.SetProjectFileDirty(projNode.IsProjectFileDirty)
+                    projNode.ComputeSourcesAndFlags()
```

Comment:
When is UpdateProjectCfg_Done called when OnAfterActiveSolutionCfgChange isn't? Are we definitely only calling ComputeSourcesAndFlags once per project on configuration change? Also is the wait dialog still open when we call ComputeSourcesAndFlags here (is this method called before OnAfterActiveSolutionCfgChange)?